### PR TITLE
Fixed #12031 - Component: Responsive Overlay

### LINKED ISSUE #12031: Component: Overlay
- [x] Implementation
- [x] Documentation

Affected Components for now;
- [x] Dropdown
- [x] MultiSelect
- [x]  TreeSelect
- [x]  CascadeSelect

### PR DESCRIPTION
**Pull request by cetincakiroglu**
New component: p-overlay

### CODE REVIEW

File: src/app/components/overlay/overlay.ts [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+]);
+
+const hideAnimation = animation([
+    animate('{{hideTransitionParams}}', style({ transform: '{{transform}}', opacity: 0 }))
+]);
+
+@Component({
+    selector: 'p-overlay',
+    template: `
+    <div #overlay [ngClass]="panelClass" [class]="panelStyleClass" [ngStyle]="panelStyle" [@overlayAnimation]="{value: 'visible', params: {transform: transformOptions, showTransitionParams: showTransitionOptions, hideTransitionParams: hideTransitionOptions}}" (click)="onOverlayClick($event)" (click)="onOverlayClick($event)">
```
Comment (volvachev) [inline src/app/components/overlay/overlay.ts:21]: duplicate event handler in template
```html
(click)="onOverlayClick($event)"
```

File: src/app/components/overlay/overlay.ts [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    
+    get overlayDirection(): string {
+        return this._overlayDirection;
+    }
+
+    @Input() set overlayDirection(value: string) {
+        const viewport = DomHandler.getViewport();
+        if (value && viewport.width < this.overlayBreakpoints) {
+            this._overlayDirection = value;
+            switch (value) {
```
Comment (volvachev) [inline src/app/components/overlay/overlay.ts:72]: Hi, mb create private method for that.

```ts
this. transformOptions = this.method(value);

//
private method(value: string): string {
        switch (value) {
            case 'start':
                return 'translate3d(0px, -100%, 0px)';
            case 'end':
                return 'translate3d(0px, 100%, 0px)';
            case 'center':
                return 'scale(0.8)';
        }
    }
```

File: src/app/components/overlay/overlay.ts [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+            this.overlayViewChild.nativeElement.style.minWidth = DomHandler.getWidth(this.container) + 'px';
+        }
+    }
+
+    alignOverlay() {
+        const viewport = DomHandler.getViewport();
+
+        if(this.overlayViewChild) {
+            if (viewport.width < this.overlayBreakpoints) {
+                switch (this.overlayDirection) {
```
Comment (volvachev) [inline src/app/components/overlay/overlay.ts:156]: also mb create method

```ts
DomHandler.addClass(this.el.nativeElement, this.getOverlayDirection(this.overlayDirection))
```

Comment (cetincakiroglu) [reply src/app/components/overlay/overlay.ts:156]: Hi @volvachev 

Thanks for your feedback, it's not the production version of the code it'll be updated and refactored before release.

**Pull request merged**